packages/lab/src/overlay/useOverlay.ts

```diff
@@ -4,128 +4,117 @@ import {
     name: "Overlay",
     state: "open",
   });
+
   const handleOpenChange = (newOpen: boolean) => {
     setOpen(newOpen);
-    onOpenChange?.(newOpen);
+    onOpenChangeProp?.(newOpen);
   };
-  const middleware = isDesktop
-    ? [margin(24), arrow({ element: arrowRef })]
-    : [
-        offset(24),
-        flip(),
-        shift({ limiter: limitShift() }),
-        arrow({ element: arrowRef }),
-      ];
+
   const {
     reference,
     floating,
     x,
     y,
     strategy,
     context,
-    middlewareData,
-    update,
+    elements,
+    middlewareData: { arrow: { x: arrowX, y: arrowY } = {} },
+    placement,
   } = useFloatingUI({
     open,
     onOpenChange: handleOpenChange,
-    placement,
-    middleware,
+    placement: placementProp,
+    middleware: [
+      offset(10),
+      flip(),
+      shift({ limiter: limitShift() }),
+      arrow({ element: arrowRef }),
+    ],
   });
 
-  const handleArrowRef = useCallback(
-    (node: HTMLDivElement) => {
-      arrowRef.current = node;
-      update();
-    },
-    [update]
-  );
-
   const { getFloatingProps, getReferenceProps } = useInteractions([
-    useDismiss(context),
     useRole(context, { role: "dialog" }),
     useClick(context),
+    useDismiss(context, {
+      outsidePressEvent: "pointerdown",
+      referencePress: true,
+    }),
   ]);
 
-  const getTriggerProps = <
-    Element extends
-      | keyof JSX.IntrinsicElements
-      | JSXElementConstructor<any> = "div"
-  >(
-    userProps?: ComponentPropsWithoutRef<Element>
-  ) => {
-    return getReferenceProps({
-      ...userProps,
-      ref: reference,
-    }) as ComponentPropsWithRef<Element>;
+  const arrowProps = {
+    ref: arrowRef,
+    context,
+    style: {
+      position: strategy,
+      left: arrowX ?? 0,
+      top: arrowY ?? 0,
+    },
   };
 
-  const getOverlayProps = (userProps?: OverlayProps) => {
-    const arrowProps = {
-      ref: handleArrowRef,
-      style: {
-        position: strategy,
-        left: middlewareData.arrow?.x ?? 0,
-        top: middlewareData.arrow?.y ?? 0,
+  const getOverlayProps = (): HTMLProps<HTMLDivElement> => {
+    return getFloatingProps({
+      // @ts-ignore
+      "data-placement": placement,
+      ref: floating,
+    });
+  };
+
+  const getTriggerProps = () =>
+    getReferenceProps({
+      ref: reference,
+      onKeyDown: (event: KeyboardEvent<HTMLElement>) => {
+        if (event.key === "Enter" || event.key === " ")
+          return handleOpenChange(false);
       },
-    };
+    });
 
-    return {
-      arrowProps,
-      open,
-      onOpenChange: handleOpenChange,
-      ...getFloatingProps({
-        // @ts-ignore
-        "data-placement": placement,
-        ...userProps,
-        style: {
-          top: y ?? 0,
-          left: x ?? 0,
-          position: strategy,
-          ...(userProps?.style || {}),
-        },
-        ref: floating,
-      }),
-    } as OverlayProps;
-  };
+  const getOverlayPosition = () => ({
```

Comment:
We should align with floating ui here and call this something like floatingStyles